File: api/routes/pdf.py
from urllib.parse import quote

from fastapi import APIRouter, HTTPException
from starlette.responses import Response
from schemas.pdf import ExportRequest

from services.pdf_service import create_resume_pdf

router = APIRouter(prefix="/api/pdf", tags=["pdf"])


@router.post("/export")
async def export_pdf(req: ExportRequest):
    try:
        pdf_bytes = await create_resume_pdf(req.resume, req.template_id)
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))

    profile = req.resume.get("profile") or {}
    name = profile.get("name", "resume")
    headline = profile.get("headline", "简历")
    filename = f"{name}_{headline}.pdf"
    encoded = quote(filename)

    return Response(
        content=pdf_bytes,
        media_type="application/pdf",
        headers={"Content-Disposition": f"attachment; filename*=UTF-8''{encoded}"},
    )

File: client/src/api/pdf.ts
import type { ResumeData } from '../types/resume'

export async function exportPdf(resume: ResumeData) {
  const response = await fetch('/api/pdf/export', {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ resume, templateId: resume.templateId })
  })
  if (!response.ok) throw new Error(await response.text())
  const blob = await response.blob()
  const url = URL.createObjectURL(blob)
  const link = document.createElement('a')
  link.href = url
  link.download = `${resume.profile.name || 'resume'}_${resume.profile.headline || '简历'}.pdf`
  link.click()
  URL.revokeObjectURL(url)
}
